feat: Add stacked segment support for PieChart sections

jay-k98 opened the pull request:
As described in https://github.com/imaNNeo/fl_chart/issues/2003 a "Stacked Pie Chart" was requested. This Pull Request aims to add this feature.

# Solution
The Implementation of [StackedPieChart] is very similar to [PieChart]. The biggest difference is drawing a Section with multiple Segments.

The radius of a segment is calculated like this:

SegmentRadius = SectionRadius * SegmentValue / TotalSectionValues

The sweep angles are then calculated like in [PieChart] 

# Todo's and Improvements
- Add drawing strokes
- Extend the Demo (add icon and more examples)
- Add tests
- Extract duplicate logic from [PieChart] and [StackedPieChart]

Bazni commented:
Hey 👋,
I think I commented at the wrong place the first time, sorry about that.

Thx for the work. I'm looking to do something like this:
<img width="247" height="234" alt="image" src="https://github.com/user-attachments/assets/8b99ce50-d40a-47c8-be72-7146357282f3" />


Do you think I will be able to do it with your StackedPieChart?

jay-k98 committed:
[ADD] stacked pie chart

Signed-off-by: Jonas Klock <jonas.klock@exxeta.com>

jay-k98 commented:
> Can you please update your branch with the latest changes? Then I can check it again. (You don't need to update the CHANGELOG.md)

Hey, I updated the branch and removed CHANGELOG.md changes.

jay-k98 committed:
[FIX] apply feedback from code-review

Signed-off-by: Jonas Klock <jonas.klock@exxeta.com>

jay-k98 commented:
@imaNNeo I get your point and I like the idea of keeping it consistent. Also as you say it would lead to a more flexible solution, which I also appreciate. I will start refactoring and hit you up when I'm done.

jay-k98 committed:
[REFAC] segments to use fromRadius and toRadius

Signed-off-by: Jonas Klock <jonas.klock@exxeta.com>

jay-k98 commented:
@imaNNeo I updated the branch and commited the refactored segment logic. I really like it because it is much cleaner and easier to understand. Thanks a lot for your feedback! Segment spacing can be solved by adding a segment. I provided that trick in the samples.

Let me know if this is what you had in mind :)

imaNNeo commented:
Nice! Thanks for the changes. I also like the new structure. Just some minor comments to solve before merging it:

imaNNeo commented on lib/src/chart/pie_chart/pie_chart_painter.dart:
```diff
@@ -175,12 +154,134 @@ class PieChartPainter extends BaseChartPainter<PieChartData> {
+  ) {
+    if (section.radius <= 0) return;
+
+    final mainPath = generateSegmentPath(
```
As I understood, we already have this path created at line 148 (called `mainPath`) in this file (the caller of this function).
So I think it's better to pass the already created path into this function and reuse it. Instead of re-creating the same path again. Please correct me if I'm wrong.

imaNNeo commented:
And please:
* Add some unit-tests for the added classes and the functionality
* Add the new property/feature in our documentation in pie_chart.md

After that, we're ready to go!

jay-k98 committed:
[REFAC] reuse main section path

Signed-off-by: Jonas Klock <jonas.klock@exxeta.com>

jay-k98 committed:
[ADD] optimize test coverage

Signed-off-by: Jonas Klock <jonas.klock@exxeta.com>

jay-k98 committed:
[ADD] documentation

Signed-off-by: Jonas Klock <jonas.klock@exxeta.com>

jay-k98 commented:
@imaNNeo Thanks for your feedback. I implemented reusing the path and added some tests for better test coverage as well as documentation. Ready for your approval :)

imaNNeo commented:
Perfect! 
Thanks for your contribution!
